### ISSUE TITLE
Syntax error from safe-auto-correct of Style/KeywordParametersOrder

### ISSUE THREAD

**Issue by esambo** [rubocop-hq/rubocop#8976]
## Problem
`rubocop --safe-auto-correct --only Style/KeywordParametersOrder` can introduce syntax error(s)!

## Expected behavior

Don't include local variable assignments as a default keyword argument during a safe auto correct.

## Actual behavior

A method without parenthesis, that has keyword arguments without defaults after keyword arguments with defaults, accidentally considers a local variable assignment on the next line as a default argument!
This could be prevented by either:
- manually moving the keyword arguments with a default to the end
- adding parenthesis around the method arguments

But a safe-auto-correct shouldn't require manual intervention.

## Steps to reproduce the problem

File: `Gemfile`
```
gem 'rubocop', '1.1.0'
```

A file with the following (contrived) code example:
`foo.rb`
```ruby
def method_name with_default: :default_value, without_default:
    local_variable = " - "
    [with_default, without_default].join(local_variable)
end
puts method_name with_default: 'Hello', without_default: 'World'
```
successfully executes:
$ `ruby foo.rb` =>
```
Hello - World
```

But a safe-auto-correct:
$ `bundle exec rubocop --safe-auto-correct --only Style/KeywordParametersOrder foo.rb` =>
```
--safe-auto-correct is deprecated; use --auto-correct
For /Users/esambo/code: Default configuration from /Users/esambo/.asdf/installs/ruby/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-1.1.0/config/default.yml
Inspecting 1 file
Scanning /Users/esambo/code/foo.rb
C

Offenses:

foo.rb:1:17: C: [Corrected] Style/KeywordParametersOrder: Place optional keyword parameters at the end of the parameters list.
def method_name with_default: :default_value, without_default:
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
Finished in 0.13004499999806285 seconds
```
changes it to:
```ruby
def method_name without_default:, with_default: :default_value    local_variable = " - "
    [with_default, without_default].join(local_variable)
end
puts method_name with_default: 'Hello', without_default: 'World'
```
which causes a syntax error (as the local variable assignment is not a default keyword argument):
$ `ruby foo.rb` =>
```
foo.rb:1: syntax error, unexpected tIDENTIFIER, expecting ';' or '\n'
...default_value    local_variable = " - "
foo.rb:3: syntax error, unexpected end, expecting end-of-input
```

## RuboCop version
$ `bundle exec rubocop -V` =>
```
1.1.0 (using Parser 2.7.2.0, rubocop-ast 1.1.0, running on ruby 2.6.6 x86_64-darwin19)
```
$ `ruby --version` =>
```
ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
```
$ `bundle --version` =>
```
Bundler version 2.1.4
```

## Related PR's
- Add new `Style/KeywordParametersOrder` cop #8563
